scripts/functions/diagnostic_environment.sh

```diff
@@ -136,59 +184,80 @@ function conda_test()
     modules_test=$?
     if [[ -z "${CONDA_PREFIX}" ]]; then printf "\n"; return 0; fi
     print_sep
-    printf "CONDA MOOSE Packages\n\n"
+    printf "CONDA MOOSE Package Checks\n\n"
     if [[ $modules_test -ge 1 ]]; then
         print_orange "WARNING: "
         printf "Unable to run Conda tests due to missing Python modules\n\n"
         return 1
     fi
     if [[ -n "${CONDA_EXE}" ]] && [[ -n "${MOOSE_NO_CODESIGN}" ]]; then
-        # right to left dependency
-        moose_packages=(dev wasp libmesh petsc)
-        conda_list=`${CONDA_EXE} list | grep 'moose-'`
+        # right to left dependency. `none` is used as a control label
+        moose_packages=(dev wasp libmesh petsc none)
+        conda_list=$(${CONDA_EXE} list ^moose)
 
         # iterate over and break on the top-most level we find installed
-        for package in ${moose_packages[@]}; do
+        for package in "${moose_packages[@]}"; do
             my_version=$(echo -e "${conda_list}" | grep "^moose-${package} " | awk '{print $2}')
             if [ -n "$my_version" ]; then
                 # hack naming conventions
-                if [[ ${package} == 'dev' ]]; then prefix='moose-'; fi
-                local needs_version=(`./versioner.py "${prefix}${package}" --yaml 2>/dev/null| \
-                                      grep 'install:' | \
-                                      awk 'BEGIN { FS = "=" }; {print $2" "$3}'`)
-                if [[ ${package} != 'dev' ]]; then
-                    reminder='moose-dev\t\t  is not present. All the packages must\n\t\t\t  kept in sync manually.'
+                if [[ ${package} == 'dev' ]]; then package='moose-dev'; fi
+
+                local needs_version
+                needs_version=()
+                while IFS='' read -r line; do needs_version+=("$line"); done < <(./versioner.py \
+                 "${package}" --yaml 2>/dev/null | \
+                 grep 'install:' | \
+                 awk 'BEGIN { FS = "=" }; {print $2}')
+
+                if [[ ${package} != 'moose-dev' ]]; then
+                    reminder='Top-level moose-dev package not in use'
                 fi
                 break
             fi
         done
-        echo -e "${conda_list}"
-        if [ -n "${reminder}" ]; then print_orange "${reminder}\n"; fi
-        if [[ -n "${my_version}" ]] && [[ "${my_version}" != "${needs_version}" ]]; then
-            print_red "\nFAIL: "
-            printf "${prefix}${package}/repository version mismatch\n"
-            printf "\nYour MOOSE repository requires moose-${package}:\t$(print_green ${needs_version})
-while your Conda environment has moose-${package}:\t$(print_red ${my_version})
+        if [[ "${package}" == 'none' ]]; then
+            printf "\nConda MOOSE packages not applicative.
```

Comment:
what do you mean by not applicative? I dont get it

Reply:
I think this `if` block was meant to reside outside of the parent `if` block. Looks like a mistake to me. I'll correct it (remove it, as I see I am doing what I want a little later with an `else`.).

Reply:
Oh wait, no... it works like this:

```bash
moose_packages=(dev wasp libmesh petsc none)
for package in moose_packages; do
   do stuff for package
done
```
if we don't find anything and we reach a literal 'none', that means the user has moose-mpi installed. And it is up to them to build petsc, libmesh, wasp manually.

Reply:
example:
```bash
conda create -n testing moose-mpi packaging jinja2 pyaml
conda activate testing
./diagnostics
<trimmed>
###############################################################################################
Python Modules (TestHarness, run-ability)

OK
###############################################################################################
CONDA MOOSE Package Checks


Conda MOOSE packages not applicative.
User is required to manually build PETSc, libMesh, and WASP


checks PASSED
```

Reply:
`Conda MOOSE packages not applicative.`
what do you mean by this? 
Maybe "conda installation path is not selected"

Reply:
But you are:

```bash
conda create -n testing moose-mpi
                        ^^^^^^^^^
```
This is basically supporting all you devs, that refuse to install `moose-petsc` or `moose-libmesh`, but you want the compiler stack.

Reply:
How about this: you tell me what you want it to say?

Like:
```pre
Conda MOOSE support library packages not installed.
```

Reply:
yeah basically. That s how I understand it. 
If it s correct then works for me

Reply:
How about this, since if we hit 'none' we no we're missing all of them:

```pre
Conda MOOSE dependencies not avilable (moose-petsc, moose-libmesh,
moose-wasp, etc). User is required to manually build PETSc, libMesh,
and WASP.
```